src/mod.ts

```diff
@@ -32,8 +32,12 @@ export const BaseProtocol = function (args: string[]): Protocol {
 /**
  * Protocol implementation that only uses stdout, but uses message boundaries to differentiate between
  * diagnostic information and hook responses.
+ * @param args command-line arguments passed to this process
+ * @returns {Protocol}
  */
-export const MessageBoundaryProtocol = function (args: string[]): Protocol {
+export const MessageBoundaryProtocol = function (
+  args: string[],
+): Required<Pick<Protocol, "getCLIFlags">> & Protocol {
```

Comment:
The base `Protocol` interface sets `getCLIFlags` as an optional key. However, the Message Boundary protocol always returns this key, so tweaking the type to make that a required property.